File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        string arr[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        set<string> m;
        for(string word: words){
            string temp;
            for(char c : word){
                temp += arr[c-'a'];
            }
            m.insert(temp);
        }
        return m.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        seen = {}
        for word in words:
            temp = ""
            for c in word:
                temp += morse[ord(c) - ord('a')]
            if temp not in seen:
                seen[temp] = 1
        return len(seen)